tracer/src/Datadog.Trace/Util/Http/HttpRequestUtils.cs

```diff
@@ -4,26 +4,46 @@
 // </copyright>
 
 using System;
-using Datadog.Trace.Configuration;
-using Datadog.Trace.Util.Http.QueryStringObfuscation;
+
+#nullable enable
 
 namespace Datadog.Trace.Util.Http
 {
     internal class HttpRequestUtils
     {
         private const string NoHostSpecified = "UNKNOWN_HOST";
 
-        internal static string GetUrl(Uri uri, QueryStringManager queryStringManager = null)
-            => GetUrl(
+        internal static string GetUrl(Uri uri, QueryStringManager? queryStringManager = null)
+        {
+            if (!uri.IsAbsoluteUri)
+            {
+                var relativeUrl = uri.ToString();
+
+                if (queryStringManager != null)
+                {
+                    var queryStringIndex = relativeUrl.IndexOf('?');
+
+                    if (queryStringIndex >= 0)
+                    {
+                        var queryString = relativeUrl.Substring(queryStringIndex);
+                        relativeUrl = relativeUrl.Substring(0, queryStringIndex) + queryStringManager.TruncateAndObfuscate(queryString);
+                    }
+                }
+
+                return relativeUrl;
```

Comment:
We should avoid doing the `ToString()` and `Substring` here, as it creates extra allocations. Can we not simplify this to:
```suggestion
                return queryStringManager == null
                    ? uri.AbsolutePath
                    : $"{uri.AbsolutePath}{queryStringManager.TruncateAndObfuscate(uri.Query)}";
```